cs/src/comm/epoxy-transport/EpoxyNetworkStream.cs

```diff
@@ -369,8 +369,8 @@ static void SafeShutdownSocket(Socket socket, Logger logger)
         {
             try
             {
-                socket.Shutdown(SocketShutdown.Both);
-                socket.Close();
+                socket?.Shutdown(SocketShutdown.Both);
+                socket?.Close();
```

Comment:
Don't we need to add this also on lines 246 and 247 or only on the safe shutdown?

Also I just realized that this method and a couple of other ones are missing the summary comments.